web/js/codeworld.js

```diff
@@ -784,9 +776,11 @@ function loadSample(code) {
 }
 
 function newProject() {
-    warnIfUnsaved(() => {
-        setCode('');
-    });
+  utils.directoryTree.clearSelectedNode()
```

Comment:
Shouldn't this be inside of warnIfUnsaved?  If I cancel the action, I don't think clearSelectedNode should be called.

Reply:
Yes, it indeed should be inside of `warnIfUnsaved`.